docs/user-manual/design-patterns/health-checking.md

```diff
@@ -0,0 +1,90 @@
+# Health Checking Pattern
+
+Certain services in flight software are critical for the correct execution of the system. For example, command dispatching is crucial to maintain control of the system. It is good practice to monitor these services to ensure they remain responsive during execution of the system. The health checking pattern is used to establish a component as a critical component of the system and periodically check it for responsiveness. 
```

Comment:
> The health checking pattern is used to establish a component as a critical component of the system and [...]

Does this design pattern actually suggest/require the component to be critical to the system? Could health checking not be used just to get some info on a non-critical component?